libbeat/common/kubernetes/types.go

```diff
@@ -98,3 +106,21 @@ type Pod struct {
 	Spec       PodSpec    `json:"spec"`
 	Status     PodStatus  `json:"status"`
 }
+
+// GetPod converts Pod to our own type
+func GetPod(pod *corev1.Pod) *Pod {
+	bytes, err := json.Marshal(pod)
+	if err != nil {
+		logp.Warn("Unable to marshal %v", pod.String())
+		return nil
+	}
+
+	po := &Pod{}
+	err = json.Unmarshal(bytes, po)
+	if err != nil {
+		logp.Warn("Unable to marshal %v", pod.String())
```

Comment:
This should stay "Unable to unmarshal". I would also suggest to make this and the above error message a bit longer to add some context from where this log message is coming from.